sdk/src/client/node_manager/mod.rs

```diff
@@ -58,6 +63,104 @@ impl std::fmt::Debug for NodeManager {
     }
 }
 
+impl ClientInner {
+    pub(crate) async fn get_request<T: DeserializeOwned + Debug + Serialize>(
+        &self,
+        path: &str,
+        query: Option<&str>,
+        timeout: Duration,
```

Comment:
I wonder if these new methods need a timeout since we have self.timeout

Reply:
made it better